tests/integration/test_upgrade.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# pylint: disable=redefined-outer-name,unused-argument,duplicate-code
+
+"""Integration test relation file."""
+
+import logging
+
+import jenkinsapi.jenkins
+import ops
+import requests
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+from pytest_operator.plugin import OpsTest
+
+import jenkins
+
+from .constants import ALL_PLUGINS
+from .helpers import get_model_jenkins_unit_address, install_plugins
+from .types_ import UnitWebClient
+
+LOGGER = logging.getLogger(__name__)
+JENKINS_APP_NAME = "jenkins-k8s-upgrade"
+
+
+async def test_jenkins_deploy_with_plugins(ops_test: OpsTest, model: Model):
+    """
+    arrange: given a juju model.
+    act: deploy Jenkins, instantiate the Jenkins client and install the plugins.
+    assert: the deployment has no errors.
+    """
+    application: Application = await model.deploy(
+        "jenkins-k8s",
+        application_name=JENKINS_APP_NAME,
+        channel="edge",
+    )
+    await application.set_config({"allowed-plugins": ",".join(ALL_PLUGINS)})
```

Comment:
This can be defined when running `model.deploy` if you specify the `config` attr